scripts/depsvalidator/Sources/DepsValidatorLibrary/Models/Manifests/Package.swift

```diff
@@ -74,6 +74,24 @@ struct Package: Decodable, SemanticValueProviding {
     }
 }
 
+extension Package.Dependency {
+    init(from decoder: Decoder) throws {
+        enum SourceControlNestedKeys: String, CodingKey {
+            case sourceControl
+        }
+        enum CodingKeys: String, CodingKey {
+            case name = "nameForTargetDependencyResolutionOnly"
+            case requirement
+        }
+        let container = try decoder.container(keyedBy: SourceControlNestedKeys.self)
+        var sourceControls = try container.nestedUnkeyedContainer(forKey: .sourceControl)
```

Comment:
```suggestion
        let sourceControls = try container.nestedUnkeyedContainer(forKey: .sourceControl)
```

Can this be a let?

Reply:
It was the first time I worked with `UnkeyedDecodingContainer`. It happened that this protocol doesn't support `for-in` syntax and uses a weird one. 
It has an internal [`currentIndex`](https://developer.apple.com/documentation/swift/unkeyeddecodingcontainer/2921359-currentindex) property and [`count`](https://developer.apple.com/documentation/swift/unkeyeddecodingcontainer/2892719-count). Every time we call `decode…` or `…Container()` methods, we are incrementing the inner index 🤯.
All this methods are mutable, so it cannot be `let`.